native/android/build.gradle

```diff
@@ -22,6 +22,8 @@ dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "net.zetetic:android-database-sqlcipher:4.4.2"
```

Comment:
It seems like we'd want to branch these based on the user's desire to use SQLCipher or not. If not, I imagine we'd want to use the native implementation.

Reply:
Yeah, I feel a little uncomfortable forcing this dependency. `android.database.sqlite.SQLiteDatabase` is built in, requires no extra code, and "just works". 

Can we easily make it so that it's optional? For example, there would be a separate .kt file which adds to `Database` support for android-database-sqlicipher, and unless we explicitly add this to app's setup we fall back to standard implementation.

I'm not super familiar with ins and outs of Android. @rozPierog ideas?

Reply:
There is no clear way to create dependency that's conditional that I know of. 
What I would do is to move and duplicate `Database.kt` to separate modules: `standard`, `cipher` and require users to include another path in `settings.gradle`+`build.gradle` to one or the other. That way person using library can have full control what is in their app.
In theory if both are included it will throw error on compile that there are two classes with same name/package
In theory if non are included it will throw error on compile that there is no such class
Database initialization would need to be handled via [Class.forName](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#forName-java.lang.String-)

To make it easier for maintenance there should be another module `common` that has common code from both Database files so that any change shouldn't be duplicated. (But I don't know how to cleanly handle type of SQLiteDatabase [` import android.database.sqlite.SQLiteDatabase` vs ` import net.sqlcipher.database.SQLiteDatabase`])
 
Disclaimer: I have no idea what I'm doing. I've never attempted stuff like this and I'm not familiar with pitfalls of modularization like that. 

@cjroth Can you give your input here?  How would you handle this?

Reply:
> Disclaimer: I have no idea what I'm doing. I've never attempted stuff like this and I'm not familiar with pitfalls of modularization like that.

Haha! Me neither, I don't know even know Kotlin :)

I think I agree with both of your comments though - I'm not aware of any way to do optional dependencies, so separating into modules seems like the best option. I do think this is kind of a bummer for users that don't need to use SQLCipher since it adds one more step to setup, but maybe it's worth it.